operator/pkg/listener/watcher_events.go

```diff
@@ -0,0 +1,147 @@
+package listener
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kyma-project/kyma-operator/operator/api/v1alpha1"
+	"github.com/kyma-project/kyma-operator/operator/pkg/labels"
+	"github.com/kyma-project/kyma-operator/operator/pkg/release"
+	"github.com/kyma-project/kyma-operator/operator/pkg/util"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type WatcherEventsHandler struct {
+	client.Client
+	logger logr.Logger
+}
+
+func (h *WatcherEventsHandler) ProcessWatcherEvent(ctx context.Context) func(event.GenericEvent, workqueue.RateLimitingInterface) {
+	h.logger = log.FromContext(ctx).WithName("skr-watcher-events-processing")
+	return func(genericEvent event.GenericEvent, q workqueue.RateLimitingInterface) {
+		//Label component CR with a current timestamp
+		//when an event is received from the SKR watcher
+		componentName := genericEvent.Object.GetName()
+		clusterName := genericEvent.Object.GetClusterName()
+		h.logger.WithValues(
+			"component", componentName,
+			"cluster-name", clusterName,
+		).Info("started dispatching event")
+
+		kymaCR, err := h.getKymaCR(ctx, clusterName)
+		if err != nil {
+			h.logger.WithValues(
+				"component", componentName,
+				"cluster-name", clusterName,
+			).Error(err, "failed to get Kyma CR for component")
+			return
+		}
+
+		componentCR, err := h.getComponentCR(ctx, kymaCR, componentName, clusterName, q)
```

Comment:
Same for the component, this failure should be handled in the reconciliation loop imo

Reply:
When you draw out all these failure conditions and just put it in the reconciliation loop I think we could achieve a better scaling for many requests since the querying of the api server will be naturally limited by the informer queue